lib/auto-languageclient.ts

```diff
@@ -517,11 +533,21 @@ export default class AutoLanguageClient {
   }
 
   // Autocomplete+ via LS completion---------------------------------------
+
+  /**
+   * A method to override to return an array of grammar scopes that should not be used for autocompletion.
+   *
+   * Usually that's used for disabling autocomplete inside comments,
+   * @example if the grammar scopes are [ '.source.js' ], `getAutocompleteDisabledScopes` may return [ '.source.js .comment' ].
+   */
+  protected getAutocompleteDisabledScopes(): Array<string> {
+    return []
+  }
+
   public provideAutocomplete(): ac.AutocompleteProvider {
     return {
-      selector: this.getGrammarScopes()
-        .map((g) => (g.includes(".") ? "." + g : g))
-        .join(", "),
+      selector: this.getGrammarScopes().join(", "),
```

Comment:
Not so sure about this change. 

cc: @illright

Reply:
Well, scopes are defined as class strings, this behaviour was sort of excusing the misuse of grammar scopes. It would allow stuff like `[ 'source.python' ]`, while actual scopes should be specified as `[ '.source.python' ]`. I guess this change is kind of breaking, but then again, the breaking would be a result of incorrectly using the scopes.

Reply:
Reverted in https://github.com/atom-community/atom-languageclient/pull/136/commits/cc6b7bf2912fe712115e2ffd3e70d812230cce31